packages/varlet-ui/src/link/Link.vue

```diff
@@ -1,7 +1,7 @@
 <template>
   <component
     :is="tag"
-    v-bind="customAttrs"
+    v-bind="linkAttrs"
```

Comment:
I think use tag `a` is enough. Because the dynamic component isn't a good choose. FYI.  i saw the implement the tag `router-link` But user can haven't install vue router. So implement a hook is better.

Reply:
Just a talk

```ts

const useRoute = () => {
  const vm = getCurrentInstance().proxy
  const router = vm.$router
  if (router) {
    return pick(router, ['push', 'replace', 'go'])
  }
  return null
}



```

Reply:
无论是使用 component 还是 hook ，如果没有装 Vue Router，最终都不会有任何效果，从结果看好像没什么区别。

如果开发者没装 Vue Router ，也没意识到这一点，页面渲染出一个裸 router-link 标签相比渲染一个 a 标签在功能不起作用下更好排查一点吧。

而且使用 hook 的话，是否会丢失浏览器对超链接的原生能力？

Reply:
> 而且使用 hook 的话，是否会丢失浏览器对超链接的原生能力？
并不会。href 可以单独判断。因为用户只会存在href和to 2个属性。to的优先级应该是最高的 它代表了 使用router的方式 这时候我们可以


```js

const internalHref = computed(() => {
      if (props.to) return 'javascript: void 0;'
      return props.href
})

```

从上面的结果来看`href`是需要进行一层包装的。`internalHref`才是最终的结果。即然我们使用了hook的方式。原生a标签的href就会被`javascript:void 0;`阻止。而我们给a标签添加的click event 就能执行。

Reply:
`href ` 的确可以单独判断，但是被赋值为 `javascript: void 0;` 后，a 标签跳转只能通过事件触发。但 a 链接本身是有一些原生能力，比如在PC端，可以拖拽链接在新 `Tab` 页打开，移动端长按链接可选择在后台打开，这些在实际开发的产品设计是很常见的需求，而使用 `hook` 行为下打开的肯定是一个空白页。

如果开发者想要保留这些能力，最终还是要自己写 `router-link` ,再套上样式，不使用 `link` 组件，这样会不会和 `link` 本身的初衷相违背。